subsys/testsuite/ztest/include/ztress.h

```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#ifndef TESTSUITE_ZTEST_INCLUDE_ZTRESS_H__
+#define TESTSUITE_ZTEST_INCLUDE_ZTRESS_H__
+
+#include <sys/util.h>
+#include <kernel.h>
+
+/** @internal Internal ID's to distinguish context type. */
+#define ZTRESS_ID_THREAD 0
+#define ZTRESS_ID_K_TIMER 1
+
+/** @brief Declare k_timer handler execution context.
+ *
+ * Context is executed in the k_timer handler context which typically means
+ * interrupt context. This context will any other used in the set.
+ *
+ * @note There can only be one k_timer context in the set and it must be the
+ * first argument of @ref ZTRESS_EXECUTE.
+ *
+ * @param handler User handler of type @ref ztress_handler.
+ *
+ * @param user_data User data passed to the @p handler.
+ *
+ * @param exec_cnt Number of handler executions to complete the test. If 0 then
+ * this is not included in completion criteria.
+ *
+ * @param init_timeout Initial backoff time base (given in @ref k_timeout_t). It
+ * is adjusted during the test to optimize CPU load. The actual timeout used for
+ * the timer is randomized.
+ */
+#define ZTRESS_TIMER(handler, user_data, exec_cnt, init_timeout) \
+	(ZTRESS_ID_K_TIMER, handler, user_data, exec_cnt, 0, init_timeout)
+
+/** @brief Declare thread execution context.
+ *
+ * Context is executed in the thread context. Priority of the thread is determined
```

Comment:
Context is executed in the thread context. -> The handler is executed within a thread context.